src/games/plinko/index.ts

```diff
@@ -99,37 +130,71 @@ export const createPlinko = ({
 export const getPlinkoResult = async ({
     coinType,
     gameSeed,
+    plinkoPackageId,
+    plinkoCorePackageId,
+    pollInterval = 3000,
+    suiClient,
     transactionResult
-}: PlinkoResultInput): Promise<PlinkoResultResponse> => {
+}: InternalPlinkoResultInput): Promise<PlinkoResultResponse> => {
     const res: PlinkoResultResponse = { ok: true };
 
     try {
         const gameInfos = getGameInfos({
             coinType,
-            corePackageId: PLINKO_CORE_PACKAGE_ID,
+            corePackageId: plinkoCorePackageId,
             gameSeed,
             moduleName: PLINKO_MODULE_NAME,
             transactionResult
         });
 
-        const settlement = await axios.post(`${DOUBLE_UP_API}/plinko`, {
-            coinType,
-            gameInfos,
-            gameName: PLINKO_MODULE_NAME
-        });
-
-        if (!settlement.data.results) {
-            throw new Error('could not determine results');
-        }
-
-        const results = [];
-
-        for (const gameResult of settlement.data.results) {
-            results.push({
-                ballCount: gameResult.ball_count,
-                betSize: gameResult.bet_size,
-                pnl: gameResult.pnl
-            });
+        let results: PlinkoResult[] = [];
+
+        while (results.length ===0) {
+            try {
+                const events = await suiClient.queryEvents({
+                    /// This is one option, another one is actually querying the event type specifically and hard coding for that.
+                    query: {
+                        MoveEventType: `${plinkoPackageId}::${PLINKO_MODULE_NAME}::Outcome<${coinType}>`
+                    },
+                    limit: 5,
```

Comment:
Arguement for making this limit 20 or even 50 because this decreases the chance we miss an event

Reply:
50 it is